SignalMessaging/attachments/SignalAttachment.swift

```diff
@@ -682,6 +682,12 @@ public class SignalAttachment: NSObject {
     private class func compressImageAsJPEG(image: UIImage, attachment: SignalAttachment, filename: String?, imageQuality: TSImageQuality) -> SignalAttachment {
         assert(attachment.error == nil)
 
+        if imageQuality == .original &&
+            attachment.dataLength < kMaxFileSizeGeneric {
```

Comment:
This will affect images picked through both "attach > document > photo library" and "attach > document > browse."

Note that I used the "generic" limit for "original/document images", not the "image" limit.